codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -172,11 +177,22 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         val targetShape =
             model.getShape(shape.target).orElseThrow { CodegenException("Shape not found: ${shape.target}") }
 
-        val targetSymbol = if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT_ZERO_VALUE_V1_NO_INPUT)) {
+        var targetSymbol = if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT)) {
             toSymbol(targetShape).toBuilder().boxed().build()
         } else {
             toSymbol(targetShape)
         }
+
+        targetSymbol = shape.getTrait<DefaultTrait>()?.let {
+            val builder = targetSymbol.toBuilder()
+            val defaultValue = it.getDefaultValue(targetShape)
+            builder.defaultValue(defaultValue)
+            if (defaultValue != "null") {
+                builder.unboxed()
+            }
+            builder.build()
+        } ?: targetSymbol
```

Comment:
**Question**: Looks like we box when the nullable index says to but then unbox if the default isn't null. Is that right? What happens if these two things conflict?